#180 - Enrich tasks with additional information

kalisp opened the pull request:
Solves: pypeclub/pype/#180

Currently task info is stored in both project config and assets as list of task names only. This PR enhances it to **list of dictionaries**.

I kept both config and asset as LISTs to follow https://github.com/getavalon/core/pull/556 and it felt wrong to have `project.config.tasks` as a dictionary and `asset.tasks' as a list.
Schemas already expect list of dictionaries, so no change is needed there.
(Could be changed easily if necessary.)

I added task name to short_name mapping. This short_name could be used in paths.. in the future.
![project_task](https://user-images.githubusercontent.com/4457962/93199394-9f479600-f74e-11ea-9409-630c6bed883a.png)
![asset_tasks](https://user-images.githubusercontent.com/4457962/93199382-9c4ca580-f74e-11ea-9db6-afd1428d2e3d.png)

Requires:
pype-config:pypeclub/pype-config#81
avalon-core:pypeclub/avalon-core#196

These queries need to be run on MongoDB:
```
db.getCollection('YOUR_COLLECTION_NAME').updateMany({type:"project"}, {$set: {"config.tasks":""}})
db.getCollection('YOUR_COLLECTION_NAME').updateMany({type:"asset"}, {$set: {"data.tasks": ""}})
```

mkolar reviewed (changes requested):
Looking at this in detail and trying out using it and retrieving data in plugins and other places, I think we should re-structure both of these to be dictionaries, rather than lists. 

With lists, we'll need to iterate and compare asset tasks against project task types everytime we need to assemble data for anatomy, which will be a major PITA. Switching to dict we could simply do 
`project_data["config"]["task_type"][asset["tasks"]["myTaskName"]["type"]]['short_code']` to get the correct short code or any other info. 

Second point is that with this we also need to adjust GUI on a few places. 
![image](https://user-images.githubusercontent.com/3333008/94119996-e5d47900-fe4f-11ea-9e76-b16140391032.png)
`class TasksModel(TreeModel)` in `avalon/tools/models` for example (maybe just there actually)

kalisp commented:
Changed do dictionaries.
![180_project](https://user-images.githubusercontent.com/4457962/94259360-a2047100-ff2e-11ea-9979-2dabaca7c24f.png)
![180_asset](https://user-images.githubusercontent.com/4457962/94259366-a466cb00-ff2e-11ea-82e3-ead3e301a374.png)

Changed everywhere I could find that it should be changed. 
Please test it in Nuke. (Maya and Clockify seems fine.)
Inventory app is not showing anything in Maya, but I don't think its a result of this change.

iLLiCiTiT commented on pype/hosts/nukestudio/tags.py:
```diff
@@ -71,8 +71,8 @@ def add_tags_from_presets():
     nks_pres_tags["[Tasks]"] = {}
-    for task in tasks:
-        nks_pres_tags["[Tasks]"][task["name"]] = {
+    for task_name, _ in tasks.items():
```
I think variable `task_name` should be `task_type` and to avoid the underscore variable you can just use `for task_name in tasks.keys():`

kalisp replied:
Modified, makes sense.
